apps/nextjs/src/app/api/chat/chatHandler.ts

```diff
@@ -179,7 +179,14 @@ async function generateChatStream(
     async () => {
       invariant(aila, "Aila instance is required");
       const result = await aila.generate({ abortController });
-      return result;
+      const transformStream = new TransformStream({
+        transform(chunk, controller) {
+          const formattedChunk = `0:${JSON.stringify(chunk)}\n`;
```

Comment:
Does `0:` here mean it's the one and only of potentially multiple simultaneous streams?

Reply:
No the stream can handle different data sources simultaneously. So if we wanted different async writers to push chunks to the stream, then we could do with a different prefix. I think the prefix of 0 is for raw text.